library/components/TCsvExport.vue

```diff
@@ -73,3 +78,12 @@ export default {
   },
 };
 </script>
+
+<style scoped>
+.disabled {
+  background-color: #f2f1f4;
+  border-color: transparent;
+  color: #b3b2bd;
+  cursor: default;
+}
+</style>
```

Comment:
Did you already try applying the styles to the disabled pseudo-class? I ask because there would be only one piece of data to use to style the button and put it into a disabled state.

e.g.
```
<button class="csv-download..."

csv-download:disabled {
  background-color: #f2f1f4;
  border-color: transparent;
  color: #b3b2bd;
  cursor: default;
}
```

I think it's also possible to do it via tailwaind, but I understand you may be doing this for better compatibility with reports-ui (to avoid all the arbitrary classes).


non-blocking:
I don't like the styling of the button very much when it's disabled. I wonder if not displaying the button would be better? You can go with this, and we can improve.